cmd/registry/patch/apply.go

```diff
@@ -20,25 +20,31 @@ import (
 	"io/fs"
 	"io/ioutil"
 	"path/filepath"
+	"strings"
 
 	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/log"
 )
 
-func Apply(ctx context.Context, client connection.Client, fileName, parent string, recursive bool) error {
-	return filepath.WalkDir(fileName,
-		func(path string, entry fs.DirEntry, err error) error {
+func Apply(ctx context.Context, client connection.Client, path, parent string, recursive bool) error {
+	return filepath.WalkDir(path,
+		func(p string, entry fs.DirEntry, err error) error {
 			if err != nil {
 				return err
-			} else if entry.IsDir() && recursive {
+			} else if entry.IsDir() && (recursive || (p == path)) {
 				return nil // Do nothing for the directory, but still walk its contents.
 			} else if entry.IsDir() {
 				return filepath.SkipDir // Skip the directory and contents.
 			}
-			return applyFile(ctx, client, path, parent)
+			return applyFile(ctx, client, p, parent)
 		})
 }
 
 func applyFile(ctx context.Context, client connection.Client, fileName, parent string) error {
+	if !strings.HasSuffix(fileName, ".yaml") {
+		return nil
+	}
+	log.FromContext(ctx).Infof("Importing %s", fileName)
```

Comment:
This seems to me like it should be part of the directory walking code. It's just another property of the file/path that we ignore during the walk.

Reply:
Thanks, looking up at the code above it, it seems that putting this check there would excessively complicate that code (and cause it to diverge from the WalkDir in the zip builder), so I'll leave this for now but am not opposed to revisiting this.